doc/source/whatsnew/v1.0.0.rst

```diff
@@ -1074,6 +1074,7 @@ Other
 - Bug in :class:`DataFrame` constructor when passing a 2D ``ndarray`` and an extension dtype (:issue:`12513`)
 - Bug in :meth:`DaataFrame.to_csv` when supplied a series with a ``dtype="string"`` and a ``na_rep``, the ``na_rep`` was being truncated to 2 characters. (:issue:`29975`)
 - Bug where :meth:`DataFrame.itertuples` would incorrectly determine whether or not namedtuples could be used for dataframes of 255 columns (:issue:`28282`)
+- Handle nested NumPy ``object`` arrays in :func:`~pandas/core/dtypes/missing.array_equivalent` (:issue:`30841`)
```

Comment:
This `:func:` link will not work (and it's also not a public function anyway), but I think you can mention it is in the pandas.testing.assert_frame/series_equal methods, as it is there eventually this array_equivalent is being used?